### ISSUE TITLE
Sign our Windows installer

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/spyder#21389]
This has been requested by a number of users to simplify and easy Spyder installation in corportate environments.

This is an initial resource on how we could sign our installer:

https://learn.microsoft.com/en-us/windows/apps/develop/smart-app-control/code-signing-for-smart-app-control

**Comment by ccordoba12** [comment]
@mrclary, could you take a look at this one? At first glance, the process seems similar to the one we use to sign our Mac app.

**Comment by mrclary** [comment]
Yep, I'll take a look. I think we can just use our Apple certificate and I think that `constructor` has the mechanism for it.

**Comment by ccordoba12** [comment]
Thanks @mrclary for your help with this!

**Comment by mrclary** [comment]
In the short term, we can use the Apple certificate, just like Napari does. I don't know anything about Windows security, but I think we will ultimately need another certificate to do it properly.

**Comment by ccordoba12** [comment]
> In the short term, we can use the Apple certificate, just like Napari does

Good idea, just to get the ball rolling. But I also think we're going to need a Windows approved certificate.

**Comment by mrclary** [comment]
@jaimergp, I noticed that Napari's most recent [release](https://github.com/napari/napari/releases/tag/v0.7.1) has a windows signed artifact. But [napari/packaging](https://github.com/napari/packaging) still indicates an Apple certificate placeholder. Can you point us in the right direction? We have windows signing credentials from NumFocus.

**Comment by jaimergp** [comment]
Keep scrolling down :D We sign _after_ constructor has run: https://github.com/napari/packaging/blob/70d6827356aa938bd5e3a982ec440efa984f4d05/.github/workflows/make_bundle_conda.yml#L587-L640

We can probably remove the Apple cert signing for Windows, but it doesn't hurt either.

**Comment by mrclary** [comment]
> Keep scrolling down :D We sign _after_ constructor has run: 

🤦🏼 I totally missed that! Thanks.

Was there something else you had to do? I get the error:
```
Attempting Azure CLI login by using service principal with secret...
Error: No subscriptions found for ***.
```

**Comment by jaimergp** [comment]
That's all we had to do. May I see your workflow? I remember some values in the 1Password note shared by NF where labeled a bit ambiguously.

**Comment by mrclary** [comment]
https://github.com/mrclary/spyder/blob/numfocus-certs/.github/workflows/installers-conda.yml#L195-L205

I've got the azure signin temporarily at the beginning of the workflow in order to fail fast. I haven't gotten past the login yet, so the signing step is still untested.

**Comment by mrclary** [comment]
I meant to update my previous comment to show the mapping that I'm using:
1password note -> action variable
* CLIENT-ID -> clientId
* VALUE -> clientSecret
* Subscription ID -> subscriptionId
* TENANT_ID -> tenantId
* URI -> endpoint
* Artifact Signing Account -> signing-account-name
* Certificate Profile -> certificate-profile-name

Also, I'm using the 1password item "Windows Application Signing - spyder"; I assume you are using "Windows Application Signing - conda". All notes values seem to be the same except for CLIENT-ID, CLIENT_SECRET_ID, and VALUE.
